### PR TITLE
SPIN-159 Add additional block device volume types to AWS pipeline deploy stage.

### PR DESCRIPTION
**Pull request by bsawyer**
@skorten @pansonm @obuckley-kenzan Please review, story is located here: https://projects.kenzan.com/jira/browse/SPIN-159

### CODE REVIEW

File: app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/blockDevices.directive.spec.js [hunk 1]
```diff
@@ -21,14 +21,26 @@ describe('Controller: BlockDevicesCtrl', function () {
...
     expect(blockDevicesCtrl.command.blockDevices.length).toBe(0);
     scope.blockDevicesCtrl.numberOfBlockDevices = 2;
     scope.$digest();
     expect(blockDevicesCtrl.command.blockDevices.length).toBe(2);
+  });
 
+  it('should set the correct min and max volume size based on volumeType', function(){
+    scope.blockDevicesCtrl.volumeType = 'sc1';
+    scope.$digest();
+    // expect(blockDevicesCtrl.volumeSizeMin).toBe(500);
```
Comment (obuckley-kenzan) [inline app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/blockDevices.directive.spec.js]: does it make sense to remove this commented out test?  Is there any way / need to test min?

Comment (bsawyer) [inline app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/blockDevices.directive.spec.js]: Will remove - hate when I leave in comments.

File: app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/advancedSettings.html [hunk 2]
```diff
@@ -1,144 +1,146 @@
...
+    </div>
+    <div class="form-group">
+      <div class="sm-label-left">
+        <b>Tags (optional)</b>
+        <help-field key="aws.serverGroup.tags"></help-field>
+      </div>
+      <map-editor model="command.tags" allow-empty="true"></map-editor>
+    </div>
+  </ng-form>
 </div>
```
Comment (pansonm) [inline app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/advancedSettings.html]: Looks like there were a lot of formatting changes here. Were these necessary/desired?

Comment (bsawyer) [inline app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/advancedSettings.html]: I had the same reaction when comparing the diff but it's the cost of adding an ng-form parent element to preserve volume size field validation. If we aren't concerned with maintaining nested element formatting I can remove the added spaces.

**Comment by pansonm** [comment]
Looks like the layout is wrapping to accommodate additional volume types (Chrome):
screen shot 2016-04-27 at 10 38 51 am

**Comment by obuckley-kenzan** [comment]
Can one of the admins verify this patch?

File: app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/advancedSettings.html (hunk 2 above)
Comment (pansonm) [inline app/scripts/modules/amazon/serverGroup/configure/wizard/advancedSettings/advancedSettings.html]: Makes sense -- didn't catch that. Looks good.

**Comment by pansonm** [comment]
There is an empty blockDevices[] array in the JSON when you set the Number of Volumes to zero. Haven't tested a deploy but best to clean up and remove array if there are no elements.

**Comment by obuckley-kenzan** [comment]
@pansonm 
I reviewed `blockDevices` field with @bsawyer .  It should be ok, there is a `command` model that reflects all the possible values and their defaults and gets updated as needed as the user fills out the form.  From there, we can see there are already existing properties with default values of `[]`.  

That said, it can't hurt to test but as far as we can tell, that is the convention already established.

**Comment by pansonm** [comment]
@obuckley-kenzan Agree this is an established pattern. I tested and it works fine.

**Pull request merged**